pkg/cmd/project/item-edit/item_edit.go

```diff
@@ -0,0 +1,286 @@
+package itemedit
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/v2/pkg/cmdutil"
+
+	"github.com/cli/cli/v2/pkg/cmd/project/shared/format"
+	"github.com/cli/cli/v2/pkg/cmd/project/shared/queries"
+
+	"github.com/cli/cli/v2/internal/tableprinter"
+	"github.com/cli/go-gh/v2/pkg/api"
+	"github.com/shurcooL/githubv4"
+	"github.com/spf13/cobra"
+)
+
+type editItemOpts struct {
+	// updateDraftIssue
+	title  string
+	body   string
+	itemID string
+	// updateItem
+	fieldID              string
+	projectID            string
+	text                 string
+	number               float32
+	date                 string
+	singleSelectOptionID string
+	iterationID          string
+	// format
+	format string
+}
+
+type editItemConfig struct {
+	tp     *tableprinter.TablePrinter
+	client *api.GraphQLClient
+	opts   editItemOpts
+}
+
+type EditProjectDraftIssue struct {
+	UpdateProjectV2DraftIssue struct {
+		DraftIssue queries.DraftIssue `graphql:"draftIssue"`
+	} `graphql:"updateProjectV2DraftIssue(input:$input)"`
+}
+
+type UpdateProjectV2FieldValue struct {
+	Update struct {
+		Item queries.ProjectItem `graphql:"projectV2Item"`
+	} `graphql:"updateProjectV2ItemFieldValue(input:$input)"`
+}
+
+func NewCmdEditItem(f *cmdutil.Factory, runF func(config editItemConfig) error) *cobra.Command {
+	opts := editItemOpts{}
+	editItemCmd := &cobra.Command{
+		Use:   "item-edit",
+		Short: "Edit an item in a project by ID",
+		Long: `
+Edit one of a draft issue or a project item. Both require the ID of the item to edit. For non-draft issues, the ID of the project is also required, and only a single field value can be updated per invocation. See the flags for more details.`,
+		Example: `
+# add --format=json to output in JSON format
+
+# edit a draft issue title and body
+gh project item-edit --id DRAFT_ISSUE_CONTENT_ID --title "a new title" --body "a new body"
```

Comment:
Since every invocation of `item-edit` has to pass the required `--id` value as it uniquely select the item to edit, could it be received as required argument instead? E.g.
```sh
gh project item-edit <item-ID> --field-id <field> --project-id <project> --text "new text"
```
We do this in other gh commands: for instance, it's `gh issue close 123`, not `gh issue close --number 123`.

Reply:
this felt more consistent with other commands that take the project number as a positional argument, for example https://github.com/cli/cli/blob/9e2e8ac441cf4fc3fc3414e929893966d37a7e9d/pkg/cmd/project/item-add/item_add.go#L42

what do you think?